### PR TITLE
feat: add PDF upload with text extraction

### PR DESCRIPTION
**Pull request by imWayneWY**
## Summary

Adds real PDF text extraction to the resume upload feature. Previously, uploading a PDF only set placeholder text (`[Uploaded: file.pdf] — PDF parsing coming soon`). Now it actually extracts the text and populates the resume textarea.

### New: `/api/parse-pdf` endpoint
- Accepts PDF file uploads via `multipart/form-data`
- Uses `pdf-parse` library to extract text from the PDF
- Validates file type (PDF only), size (max 10MB), and non-empty content
- Returns extracted text as JSON (`{ text: "..." }`)
- Returns appropriate errors for corrupted, password-protected, or image-based PDFs (422)
- Returns validation errors for wrong file type, empty files, oversized files (400)

### Updated: Tailor page (`/tailor`)
- `processFile` now sends the PDF to `/api/parse-pdf` and populates the textarea with extracted text
- Added `isParsing` loading state with spinner and "Extracting text from PDF…" message
- Disabled drag/drop zone and file input during parsing
- Submit button is disabled during PDF parsing
- Graceful error handling for API errors and network failures
- **Manual text paste still works** — PDF upload is an additional option, not a replacement

### Tests
- **9 new tests** for the `/api/parse-pdf` route (validation, success, errors, edge cases)
- **4 new tests** for the tailor page (successful extraction, API errors, network errors, loading state)
- Updated 1 existing test to reflect new behavior (previously checked for placeholder text)
- **All 146 tests pass** ✅

## Dependencies
- Added `pdf-parse` (runtime)
- Added `@types/pdf-parse` (dev)

### CODE REVIEW

File: src/app/api/parse-pdf/route.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        { error: "Could not extract any text from the PDF. The file may be scanned or image-based." },
+        { status: 422 }
+      );
+    }
+
+    return NextResponse.json({ text });
+  } catch {
+    return NextResponse.json(
+      { error: "Failed to parse PDF. The file may be corrupted or password-protected." },
+      { status: 422 }
```
[the remark this answers is not in the document]
Comment (imWayneWY) [reply src/app/api/parse-pdf/route.ts]: Fixed — changed to 500 with console.error logging. The generic message avoids misleading clients, and the logged error helps with debugging.

File: src/app/api/parse-pdf/__tests__/route.test.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+// ---------- tests ----------
+
+describe("POST /api/parse-pdf", () => {
+  beforeEach(() => {
+    mockPdfParse.mockReset();
+  });
+
+  it("returns 400 when no file is provided", async () => {
+    const req = makeFormDataRequest(null);
```
[the remark this answers is not in the document]
Comment (imWayneWY) [reply src/app/api/parse-pdf/__tests__/route.test.ts]: Already cleaned up in the fix commit — the unused declaration has been removed.
